apps/signal/56-forms-and-signal/src/app/order.component.ts

```diff
@@ -55,18 +57,22 @@ import { products } from './products';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export default class OrderComponent {
+  private readonly route = inject(ActivatedRoute);
   form = new FormGroup({
-    quantity: new FormControl(1, { nonNullable: true }),
+    quantity: new FormControl(
+      +(this.route.snapshot.queryParams['quantity'] ?? 1),
+      { nonNullable: true },
+    ),
   });
 
-  productId = input('1');
+  productId = input.required({ transform: numberAttribute });
   price = computed(
     () => products.find((p) => p.id === this.productId())?.price ?? 0,
   );
   quantity = toSignal(this.form.controls.quantity.valueChanges, {
     initialValue: this.form.getRawValue().quantity,
   });
-  totalWihoutVat = computed(() => Number(this.price()) * this.quantity());
-  vat = computed(() => this.totalWihoutVat() * 0.21);
-  total = computed(() => this.totalWihoutVat() + this.vat());
+  totalWithoutVat = computed(() => Number(this.price()) * this.quantity());
```

Comment:
you should have fixed this typo in a PR 😇

Reply:
Good point. I just submitted that here: https://github.com/tomalaforge/angular-challenges/pull/1126